src/text/scrptrun.cpp

```diff
@@ -156,8 +160,9 @@ UBool ScriptRun::next()
         // characters above it on the stack will be poped.
         if (pairIndex >= 0) {
             if ((pairIndex & 1) == 0) {
-                parenStack[++parenSP].pairIndex = pairIndex;
-                parenStack[parenSP].scriptCode  = scriptCode;
+                ++parenSP;
+                parenStack.emplace_back(pairIndex, scriptCode);
```

Comment:
This looks wrong. How do I construct a test case that'd fail because of this?
There is no corresponding `pop_back` anywhere in the code. Popping or clearing the stack is done by simply decreasing `parenSP`, but the vector maintains its size, and so items emplaced after a pop will not be at position `parenSP`. Either there need to be actual vector pops everywhere `parenSP` decreases (and consequently `parenSP` made redundant as it will always be equal to `parenStack.size() - 1`), or keep this simplistic raw-array-in-a-vector approach by only pushing to vector when `parenSP >= size()`
```c++
++parenSP;
if ((size_t)parenSP < parenStack.size())
    parenStack[parenSP] = { pairIndex, scriptCode };
else
    parenStack.emplace_back(pairIndex, scriptCode);
```

Reply:
Good catch ^ thanks @lightmare !

Reply:
https://github.com/mapnik/mapnik/commit/9dac8dcc34f57f6529219b28f06d05185c24840b